src/helpers/generateField.ts

```diff
@@ -32,10 +36,26 @@ export const generateField = (
 
   if (list) fieldType = ts.factory.createArrayTypeNode(fieldType);
 
-  return ts.factory.createPropertySignature(
+  const propertySignature = ts.factory.createPropertySignature(
     undefined,
     ts.factory.createIdentifier(name),
     undefined,
     fieldType
   );
+
+  if (documentation) {
+    return ts.addSyntheticLeadingComment(
+      propertySignature,
+      ts.SyntaxKind.MultiLineCommentTrivia,
+      // This is a bit hacky and I'm not sure if there's a better way to do this
+      // while returning a PropertySignature type.
+      // This methods results to: /*${documentation}*/
+      // Which is clearly not a JSDoc comment. So to get around it, we add a
+      // leading asterisk + surrounding spaces so the result is: /** ${documentation} */
+      `* ${documentation} `,
```

Comment:
I think this function should do the trick:
```ts
export const formatJSDocComment = (comment: string) => {
  return `*\n * ${comment.split("\n").join("\n * ")}\n `;
};
```

It could live in `src/utils/jsdoc.ts` or something like that. Then formatting should be as easy as:

```suggestion
      formatJSDocComment(documentation),
```

Reply:
Made a util function as per suggestion. Called it `applyJSDocWorkaround` instead. `formatJSDocComment` doesn't sit well with me when it's not doing that exactly and quite misleading, thoughts? 🤣

Reply:
Is it a workaround though? Aren't we just applying formatting to the comment so that it fits the JS doc spec? I'm ambivalent though, so its up to you. I'll resolve the conversation and approve the PR. 🆒

Reply:
@zeferinix Ready to merge once you add a changeset!

Reply:
Will push shortly!

Reply:
Pushed @valtyr! Lemme know if that works. Not exactly sure if this should've been a patch 😆